spec/models/email_authentication_spec.rb

```diff
@@ -1,47 +1,89 @@
-      auth.password = password
-      expect(auth).to_not be_valid, "password #{password} should not be valid"
-      expect(auth.errors.messages[:password]).to eq [weak_password_error]
+    it "can't be blank" do
+      auth = build(:email_authentication, password: "")
+      expect(auth).to_not be_valid
+      # expect(auth.errors.messages[:password]).to include "can't be blank"
+    end
+
+    it "requires at least ten characters" do
+      auth = build(:email_authentication, password: "a" * 9)
+      expect(auth).to_not be_valid
+      # expect(auth.errors.messages[:password]).to include "must be between 10 and 128 characters"
+    end
+
+    it "requires fewer than 128 characters" do
+      auth = build(:email_authentication, password: "a" * 129)
+      expect(auth).to_not be_valid
+      # expect(auth.errors.messages[:password]).to include "must be between 10 and 128 characters"
+    end
+
+    it "requires at least one lower case letter" do
+      auth = build(:email_authentication, password: "A" * 10)
+      expect(auth).to_not be_valid
+      # expect(auth.errors.messages[:password]).to include "must contain at least one lower case letter"
     end
-  end
 
-  it "allows strong passwords" do
-    auth = build(:email_authentication)
-    ["three word passphrase", "stubborn kosher rehire", "brown batter horse pumpkin",
-      "speaker imac coverage flower", "@zadlfj4809574zk.vd", "long-pass-phrase-verklempt-basketball"].each do |password|
-      auth.password = password
-      expect(auth).to be_valid, "password #{password.inspect} should not be valid"
+    it "requires at least one upper case letter" do
+      auth = build(:email_authentication, password: "a" * 10)
+      expect(auth).to_not be_valid
+      # expect(auth.errors.messages[:password]).to include "must contain at least one upper case letter"
+    end
+
+    it "requires at least one number" do
+      auth = build(:email_authentication, password: "NoNumbers!")
+      expect(auth).to_not be_valid
+      # expect(auth.errors.messages[:password]).to include "must contain at least one number"
+    end
+
+    it "is valid when it is at least ten characters and includes a number and lower and upper case letters" do
+      auth = build(:email_authentication, password: "V4lid8me!!!")
+      expect(auth).to be_valid
     end
   end
 
-  it "allows email to be updated even if the EmailAuthentication has a weak password" do
-    auth = build(:email_authentication, password: "1234567890")
-    auth.save(validate: false)
-    auth = EmailAuthentication.find(auth.id)
-    auth.email = "newemail@example.com"
-    expect(auth).to be_valid
-    auth.save!
-    # make sure a weak password change is prevented
-    auth.password = "passw0rd"
-    expect(auth).to_not be_valid
-    expect(auth.errors.messages[:password]).to eq [weak_password_error]
+  describe "updating" do
+    it "does not allow password to be an nil" do
```

Comment:
I'm specifically testing this case because there is no model validation for presence of password on update. That is inherent to the design, since we need to allow it to be nil in order to update the email without the password. A nil password is guarded against on update by either devise or the postgres.